### DISCUSSION TITLE
Does laspy change crs when reading the file

### DISCUSSION THREAD

**Discussion by nealresearch** [laspy/laspy#259] [Q&A] [answered]
Whenever i read a las data. The coordinate is different from what it is suppose to be. For example if i read the data using lidr in r then i get this coordinate (368230.8, 5807508). But when i used laspy then i get this weird coordinate (286999880, 191722170)

This is how i read the las data

inFile = laspy.read ("lidar.laz")
print(inFile.X[1],infile.Y[1])

**Comment by zwnk** [comment, marked as answer]
```python
# Grab just the X dimension from the file, and scale it.

def scaled_x_dimension(las_file):
    x_dimension = las_file.X
    scale = las_file.header.scales[0]
    offset = las_file.header.offsets[0]
    return (x_dimension * scale) + offset

scaled_x = scaled_x_dimension(las)
```
las_file.X has to be multiplied by the scale factor and added the offset to get the valid coordinate. 
see here: https://laspy.readthedocs.io/en/latest/complete_tutorial.html

**Reply by tmontaigu** [reply]
And the `las_file.x`,  `las_file.y` `las_file.z` does this for you
